photon-ml/src/main/scala/com/linkedin/photon/ml/optimization/game/RandomEffectOptimizationTracker.scala

```diff
@@ -53,24 +55,61 @@ protected[ml] class RandomEffectOptimizationTracker(val optimizationStatesTracke
   }
 
   /**
-   * Build a summary string for the tracker
+   * Count the number of occurrence of all returned convergence reasons
    *
-   * @return string representation
+   * @return A map of convergence reason to its count of occurrence from the random effect optimization problems
    */
-  override def toSummaryString: String = {
-    val convergenceReasons = optimizationStatesTrackers.map { optimizationStatesTracker =>
+  def countConvergenceReasons: Map[String, Int] = {
+    optimizationStatesTrackers.map { optimizationStatesTracker =>
       if (optimizationStatesTracker.convergenceReason.isDefined) {
         (optimizationStatesTracker.convergenceReason.get.reason, 1)
       } else {
-        ("Not converged", 1)
+        (RandomEffectOptimizationTracker.NOT_CONVERGED, 1)
+      }
+    }.reduceByKey(_ + _).collectAsMap()
+  }
+
+  /**
+   * Get stats counter of the number of iterations tracked in all optimization states trackers
+   *
+   * @return A [[StatCounter]] of the number of iterations tracked in all optimization states trackers
+   */
+  def getNumIterationStats: StatCounter = {
+    optimizationStatesTrackers.map(_.getTrackedStates.length).stats()
+  }
+
+  /**
+   * Get stats counter on the elapsed time tracked in all optimization states trackers
+   *
+   * @return A [[StatCounter]] of the elapsed time tracked in all optimization states trackers
+   */
+  def getElapsedTimeStats: StatCounter = {
+    optimizationStatesTrackers.map { optimizationStatesTracker =>
+      val timeHistory = optimizationStatesTracker.getTrackedTimeHistory
+      if (timeHistory.length > 0) {
+        timeHistory.last * 1e-6
+      } else {
+        0.0
       }
-    }.reduceByKey(_ + _).collectAsMap().mkString("\n")
-    val numIterationsStats = optimizationStatesTrackers.map(_.getTrackedTimeHistory.length).stats()
-    val timeElapsedStats = optimizationStatesTrackers.map { optimizationStatesTracker =>
-      optimizationStatesTracker.getTrackedTimeHistory.last * 1e-6
     }.stats()
```

Comment:
Calling ```last``` directly is problematic when the length of the array returned by ```optimizationStatesTracker.getTrackedTimeHistory``` is empty, e.g., when no optimization states is tracked.